src/configuration/index.js

```diff
@@ -0,0 +1,205 @@
+import Promise from 'pinkie';
+import { fsObjectExists, readFile } from '../utils/promisified-functions';
+import Option from './option';
+import optionSource from './option-source';
+import { cloneDeep, castArray } from 'lodash';
+import { ensureOptionValue as ensureSslOptionValue } from '../utils/parse-ssl-options';
+import OPTION_NAMES from './option-names';
+import { optionValueToRegExp } from './option-conversion';
+import createFilterFn from '../utils/create-filter-fn';
+import resolvePathRelativelyCwd from '../utils/resolve-path-relatively-cwd';
+
+const CONFIGURATION_FILENAME = '.testcaferc.json';
+
+const ERR_READ_CONFIG_FILE                 = 'An error is occurred during reading the configuration filename.';
+const ERR_CONFIG_FILE_IS_NOT_WELLFORMATTED = 'Configuration filename is not well-formatted.';
```

Comment:
I think this message doesn't reflect that json is can't be parsed. Probably it suppose `well-formed`, but anyway it should describe expected format (json is not well-formed).

Reply:
Ok. In this case, I propose to use this message:
```js
Configuration filename is not well-formed.
Failed to parse '.testcaferc.json' data.
.testcaferc.json must be actual JSON.
```
This message is similar with
![image](https://user-images.githubusercontent.com/4133518/49874872-b1d1c480-fe30-11e8-910f-c67381d22c78.png)